static/app/views/performance/transactionSummary/transactionVitals/index.spec.jsx

```diff
@@ -149,203 +159,168 @@ describe('Performance > Web Vitals', function () {
+      context: routerContext,
+    });
+    expect(screen.getByText("You don't have access to this feature")).toBeInTheDocument();
   });
 
-  it('renders the basic UI components', async function () {
-    const {organization, router, routerContext} = initialize();
+  it('renders the basic UI components', function () {
+    const {organization, router, routerContext} = initialize({
+      transaction: '/organizations/:orgId/',
+    });
 
-    const wrapper = mountWithTheme(
+    render(
       <WrappedComponent
         organization={organization}
         location={router.location}
         router={router}
       />,
-      routerContext
+      {context: routerContext}
     );
 
-    await tick();
-    wrapper.update();
+    expect(
+      screen.getByRole('heading', {name: '/organizations/:orgId/'})
+    ).toBeInTheDocument();
 
-    expect(wrapper.find('TransactionHeader')).toHaveLength(1);
-    expect(wrapper.find('SearchBar')).toHaveLength(1);
-    expect(wrapper.find('TransactionVitals')).toHaveLength(1);
+    ['navigation', 'main'].forEach(role => {
+      expect(screen.getByRole(role)).toBeInTheDocument();
+    });
   });
 
-  it('renders the correct bread crumbs', async function () {
+  it('renders the correct bread crumbs', function () {
     const {organization, router, routerContext} = initialize();
 
-    const wrapper = mountWithTheme(
+    render(
       <WrappedComponent
         organization={organization}
         location={router.location}
         router={router}
       />,
-      routerContext
+      {context: routerContext}
     );
 
-    await tick();
-    wrapper.update();
-
-    expect(wrapper.find('Breadcrumb').text()).toEqual(
-      expect.stringContaining('Web Vitals')
-    );
+    expect(screen.getByRole('navigation')).toHaveTextContent('PerformanceWeb Vitals');
   });
 
-  it('renders all vitals cards correctly', async function () {
+  describe('renders all vitals cards correctly', function () {
     const {organization, router, routerContext} = initialize();
 
-    const wrapper = mountWithTheme(
-      <WrappedComponent
-        organization={organization}
-        location={router.location}
-        router={router}
-      />,
-      routerContext
-    );
-
-    await tick();
-    wrapper.update();
-
-    const vitalCards = wrapper.find('VitalCard');
-    expect(vitalCards).toHaveLength(5);
-
-    vitalCards.forEach((vitalCard, i) => {
-      expect(vitalCard.find('CardSectionHeading').text()).toEqual(
-        expect.stringContaining(vitals[i].heading)
+    beforeEach(() => {
+      render(
+        <WrappedComponent
+          organization={organization}
+          location={router.location}
+          router={router}
+        />,
+        {context: routerContext}
       );
-      expect(vitalCard.find('StatNumber').text()).toEqual(vitals[i].baseline);
     });
-    expect(vitalCards.find('BarChart')).toHaveLength(5);
+
+    it.each(vitals)('Renders %s', function (vital) {
```

Comment:
I don't really like this pattern myself, it's a bit wasteful since the component gets mounted and unmounted between each expect

Reply:
But the failures are so nice—you get a failure for the exact case that failed with a descriptive message! I agree that the mounting feels wasteful, though. I tried to use `beforeAll` first, but that runs before `beforeEach`, so the mocks weren't ready. Then I convinced myself that remounting before every test is actually a good thing 🤔

Reply:
Not a big deal either way 🤷

Reply:
Unless you are dealing with massive render times I think a more ergonomic test beats out perf wins, when in doubt I'd just profile it when running jest and confirm the rerenders are negligible. Or better yet just merge it and check the transaction times once it reaches our sentry jest project in the perf product 😄, since CI time is mostly what we're optimizing for.

Reply:
as-in I would optimize this for perf only if perf problems were a large enough concern (eg. using `user.type` with large input 😆) but otherwise 👍

Reply:
Alright, I'm going to merge then! There's a bunch of `getByRole` here, but replacing it with `getByTestId` did not make a meaningful difference 👌🏻 Nothing else should have a major performance hit anyway